internal/artifacts/module_arch_test.go

```diff
@@ -980,8 +1015,46 @@ module-types:
 			Ω(selectedModulesSorted).Should(Equal([]string{"m1", "n1"}))
 		})
 	})
+
+	Describe("buildSelectedModule", func() {
+
+		projectFolder := getTestPath("result", "mta_with_flat_module")
+
+		BeforeEach(func() {
+			Ω(dir.CopyDir(getTestPath("mta_with_flat_module"), projectFolder, true, dir.CopyEntries)).Should(Succeed())
+		})
+
+		It("ignores default target folder when target is not defined and default is subfolder of packaged module content", func() {
+			// build first module to create some content in the default target folder
+			Ω(buildSelectedModule(projectFolder, "", nil, "mod1", true, make(map[string]string), getPathToFlatModule)).Should(Succeed())
+			module1ZipPath := getTestPath("result", "mta_with_flat_module", ".mta_with_flat_module_mta_build_tmp", "mod1", "data.zip")
+			Ω(module1ZipPath).Should(BeAnExistingFile())
+			// build second module whose content is the whole project
+			Ω(buildSelectedModule(projectFolder, "", nil, "mod2", true, make(map[string]string), getPathToFlatModule)).Should(Succeed())
+			module2ZipPath := getTestPath("result", "mta_with_flat_module", ".mta_with_flat_module_mta_build_tmp", "mod2", "data.zip")
+			Ω(module2ZipPath).Should(BeAnExistingFile())
+			validateArchiveContents([]string{"test.txt", "sub1/test1.txt", "mta.yaml", "sub1/", "sub2/", "sub2/test2.txt"}, module2ZipPath)
+		})
+
+		It("ignores specified target folder when target is defined and target is subfolder of packaged module content", func() {
+			targetFolder := getTestPath("result", "mta_with_flat_module", "target")
+			// build first module to create some content in the specified target folder
+			Ω(buildSelectedModule(projectFolder, targetFolder, nil, "mod1", true, make(map[string]string), getPathToFlatModule)).Should(Succeed())
+			module1ZipPath := getTestPath("result", "mta_with_flat_module", "target", "data.zip")
+			Ω(module1ZipPath).Should(BeAnExistingFile())
+			// build second module whose content is the whole project
+			Ω(buildSelectedModule(projectFolder, targetFolder, nil, "mod2", true, make(map[string]string), getPathToFlatModule)).Should(Succeed())
+			module2ZipPath := getTestPath("result", "mta_with_flat_module", "target", "data.zip")
+			Ω(module2ZipPath).Should(BeAnExistingFile())
+			validateArchiveContents([]string{"test.txt", "sub1/test1.txt", "mta.yaml", "sub1/", "sub2/test2.txt", "sub2/"}, module2ZipPath)
+		})
+	})
 })
 
+func getPathToFlatModule() (string, error) {
```

Comment:
Why does this return an error if it's always `nil`?

Reply:
because it has to match the signature of os.Getwd function

Reply:
Add a comment on this function that it's used to simulate building in the project working directory, because the default build result folder in module build is relative to the current working directory

Reply:
added